First pass at README (#17)
dell/csm-sharednfs: README.md

[![Docker](https://img.shields.io/docker/pulls/dellemc/csm-hbnfs.svg?logo=docker&style=flat-square&label=Pulls)](https://hub.docker.com/r/dellemc/csm-hbnfs)
[![Last Release](https://img.shields.io/github/v/release/dell/csm-hbnfs?label=Latest&style=flat-square&logo=go)](https://github.com/dell/csm-hbnfs/releases)

## Description
The csm-hbnfs (Container Storage Modules - Host-Based NFS) is a component designed to export block volumes from a CSI driver via NFS. It enhances Kubernetes environments by providing NFS access to block storage volumes, allowing multiple pods to share storage efficiently.

## Features
    Converts block storage into NFS shares. 
    Enables multiple pods to access the same volume. 
    Integrates seamlessly with Dell Storage solutions. 
    Supports dynamic volume provisioning. Optimized for Kubernetes environments.

## Usage
Once deployed, csm-hbnfs allows users to: 
    Create and manage NFS shares backed by block storage. 
    Mount NFS volumes on multiple pods. 
    Ensure high availability and performance in storage workloads. 

>     This project can be compiled with CSI Powerstore driver at this time. Other platforms are not yet supported at this time.

Reviewer: suggestion: improve formatting and change the last bullet to a note.
```suggestion
Once deployed, csm-hbnfs allows users to: 
    - Create and manage NFS shares backed by block storage. 
    - Mount NFS volumes on multiple pods. 
    - Ensure high availability and performance in storage workloads. 
    
> Note: This project can be compiled with CSI Powerstore driver at this time. Other platforms are not yet supported at this time.
```

## Table of Contents

* [Code of Conduct](https://github.com/dell/csm/blob/main/docs/CODE_OF_CONDUCT.md)
* [Maintainer Guide](https://github.com/dell/csm/blob/main/docs/MAINTAINER_GUIDE.md)
* [Committer Guide](https://github.com/dell/csm/blob/main/docs/COMMITTER_GUIDE.md)
* [Contributing Guide](https://github.com/dell/csm/blob/main/docs/CONTRIBUTING.md)
* [List of Adopters](https://github.com/dell/csm/blob/main/docs/ADOPTERS.md)
* [Support](#support)
* [Security](https://github.com/dell/csm/blob/main/docs/SECURITY.md)
* [Building](#building)
* [Runtime Dependecies](#runtime-dependencies)
* [Driver Installation](#driver-installation)
* [Using Driver](#using-driver)
* [Documentation](#documentation)

## Support
For any issues, questions or feedback, please follow our [support process](https://github.com/dell/csm/blob/main/docs/SUPPORT.md)

## Building

> This project is imported as a Go module (see golang.org Module information for explanation) from within the CSI drivers.

Reviewer: suggestion: could provide a hyperlink for easier reading?
```suggestion
This project is imported as a [Go module](https://go.dev/ref/mod) from within the CSI drivers.
```

The dependencies for this project are listed in the go.mod file.

> To run unit tests, execute `make test`.

Reviewer: suggestion: needs to be `make unit-test`, or makefile should be updated to change the target to `test`.
```suggestion
To run unit tests, execute `make unit-test`.
```

# CSM Host Based NFS for Dell CSI drivers

[![Go Report Card](https://goreportcard.com/badge/github.com/dell/csm-hbnfs?style=flat-square)](https://goreportcard.com/report/github.com/dell/csm-hbnfs)
[![License](https://img.shields.io/github/license/dell/csm-hbnfs?style=flat-square&color=blue&label=License)](https://github.com/dell/csm-hbnfs/blob/master/LICENSE)

> [![Docker](https://img.shields.io/docker/pulls/dellemc/csm-hbnfs.svg?logo=docker&style=flat-square&label=Pulls)](https://hub.docker.com/r/dellemc/csm-hbnfs)

Reviewer 2: Can we check any for quay.io?

Author: Good point. I think we are not publishing any csm-hbnfs images, so maybe get rid of the docker line?

Author: Removed reference to docker as we are not publishing any images for hbnfs.

[![Last Release](https://img.shields.io/github/v/release/dell/csm-hbnfs?label=Latest&style=flat-square&logo=go)](https://github.com/dell/csm-hbnfs/releases)

## Description
The csm-hbnfs (Container Storage Modules - Host-Based NFS) is a component designed to export block volumes from a CSI driver via NFS. It enhances container orchestrator environments by providing NFS access to block storage volumes, allowing multiple pods to share storage efficiently.

## Features
    Converts block storage into NFS shares. 
    Enables multiple pods to access the same volume. 
    Integrates seamlessly with Dell Storage solutions. 
    Supports dynamic volume provisioning. Optimized for Kubernetes environments.

## Usage
Once deployed, csm-hbnfs allows users to: 
    - Create and manage NFS shares backed by block storage. 
    - Mount NFS volumes on multiple pods. 

>     - Ensure high availability and performance in storage workloads. 

Reviewer: It looks like it still didn't get list formatting correct. Try removing the leading empty space. I think that should fix it.

Author: How are you checking?

Reviewer: in VSCode you can use `ctrl+shift+v` or on github, you can view click "view file" from the dropdown in the upper right hand corner of the changes in the PR.
![image](https://github.com/user-attachments/assets/e6357e48-c99a-41a6-86c8-4979b02a9880)

Reviewer: you can also edit it here, on GitHub and there is a preview tab

Author: OK, gotcha.

Author: done

Note: This project can be compiled with CSI Powerstore driver only. Other platforms are not yet supported at this time.

## Table of Contents

* [Code of Conduct](https://github.com/dell/csm/blob/main/docs/CODE_OF_CONDUCT.md)
* [Maintainer Guide](https://github.com/dell/csm/blob/main/docs/MAINTAINER_GUIDE.md)
* [Committer Guide](https://github.com/dell/csm/blob/main/docs/COMMITTER_GUIDE.md)
* [Contributing Guide](https://github.com/dell/csm/blob/main/docs/CONTRIBUTING.md)
* [List of Adopters](https://github.com/dell/csm/blob/main/docs/ADOPTERS.md)
* [Support](#support)
* [Security](https://github.com/dell/csm/blob/main/docs/SECURITY.md)
* [Building](#building)

> * [Runtime Dependecies](#runtime-dependencies)

Reviewer 2: Do we have runtime Dependecies ?
Spelling

Author: Good point. I think we are not publishing any csm-hbnfs images, so maybe get rid of the docker line?
